baseplate/core.py

```diff
@@ -123,6 +124,123 @@ def from_upstream(cls, trace_id, parent_id, span_id, sampled, flags):
         return cls(trace_id, parent_id, span_id, sampled, flags)
 
 
+class AuthenticationContextFactory(object):
```

Comment:
:dizzy_face: I think I've ended up thinking this is good as is but totally see the confusion potential of `ContextFactory`